### ISSUE TITLE
Alerting: Provisioning API should validate deleteRules directive

### ISSUE THREAD

**Issue by darks00ul** [grafana/grafana#61885]
**What happened**:
I have a Grafana lab enviroment where I'm playing around with Grafana Provisioning.
I noted that if I create a YAML containing deleteRules declared with an orgId but without any uid, it allows me to modify or delete provisioned alert rules belongin to that orgId. 
If I reload Grafana all rules are restored again.

**What you expected to happen**:
Provisioned rules should not be editable/erasable

**How to reproduce it (as minimally and precisely as possible)**:
create a YAML file at ./provisioning/alert with the following lines: 

```
apiVersion: 1
deleteRules:
  - orgId: 1
#    uid: xxxxxxx
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.3.2
- Data source type & version: Prometheus 
- OS Grafana is installed on: Ubuntu 22.04
- User OS & Browser: Brave 1.47.171 - Windows 10
- Grafana plugins: default config
- Others:

**Comment by zuchka** [comment]
Thanks for creating this issue, @darks00ul 

so you are provisioning alert rules in one file and then attempting to delete those rules in another file?

**Comment by darks00ul** [comment]
> Thanks for creating this issue, @darks00ul
> 
> so you are provisioning alert rules in one file and then attempting to delete those rules in another file?

Thanks for your reply zuchka.

About your question, I start grafana with some alert rules provisioned. 

For convenience I kept the deleteRule yaml in my alert folder but without any alert uid defined.

It was just a coincidence that I found out that the alert rules were created but without the "provisioned" message in grafana, so I was able to modify or delete these alert rules but on every Grafana restore, these rules are restored.

As far I understand, this is not the expected behavior. So this an undocumented "feature" or a bug :D
